ext/mbstring/php_mbregex.c

```diff
@@ -1400,6 +1400,11 @@ PHP_FUNCTION(mb_ereg_search_setpos)
 		return;
 	}
 
+	/* Accept negative position if length of search string can be determined */
+	if ((position < 0) && (!Z_ISUNDEF(MBREX(search_str))) && (Z_TYPE(MBREX(search_str)) == IS_STRING)) {
+		position += Z_STRLEN(MBREX(search_str));
```

Comment:
Yes, we could delay computing the final offset but it would also delay the potential 'out of range' message.